File: codeforce/1552E.py
#greedy
import math
n,k=list(map(int, input().split()))
a,b=[0]*(n+1),[0]*(n+1)

c=[ [] for _ in range(1+n) ]

nk=n*k
mxcnt=math.ceil(n/(k-1))

for i,clr in enumerate( list(map(int, input().split())) ):
    c[clr].append(i+1)

it=[ [] for _ in range(k) ]

for i in range(1,1+n):
    for j in range(1,k):
        it[j].append( (c[i][j], c[i][j-1], i) )

remain=n
step=1

while remain>0:
    cnt=0
    for e,s,clr in sorted(it[step]):
        if a[clr]>0:
            continue
        a[clr],b[clr]=s,e
        cnt+=1
        remain-=1
        if cnt==mxcnt:
            break
    step+=1

for i in range(1,n+1):
    print("%d %d\n"%(a[i],b[i]))

File: codeforce/1552E.cpp
//greedy
#include <stdio.h>
#include <vector>
#include <algorithm>
#include <tuple>
using namespace std;
int a[101], b[101];
int cnt[10001];
int main() {
	vector<int> c[101];
	vector<tuple<int,int,int> > it;
	int i, j, n, k; scanf("%d%d",&n,&k);
	int nk = n * k;
	int mxcnt = n / (k - 1) + ( (n%(k-1))>0);
	for (i = 1; i <= nk; ++i) {
		int clr; scanf("%d",&clr);
		c[clr].push_back(i);
	}
	for (i = 1; i <= n; ++i) {
		for (j = 1; j < k; ++j)
			it.push_back({c[i][j],c[i][j-1],i});
	}
	sort(it.begin(), it.end());
	for (auto range : it) {
		int s, e, clr; tie(e, s, clr) = range;
		if (a[clr] != 0) continue;
		int mx=0;
		for (i = s; i <= e && mx<=mxcnt; ++i) 
			mx = max(mx, cnt[i]+1);
		if (mx <= mxcnt) {
			a[clr] = s; b[clr] = e;
			for (i = s; i <= e; ++i) ++cnt[i];
		}
	}

	for (i = 1; i <= n; ++i)
		printf("%d %d\n",a[i],b[i]);


	return 0;
}